openverse_catalog/dags/providers/provider_api_scripts/europeana.py

```diff
@@ -8,266 +8,172 @@
 
-def _extract_image_list_from_json(response_json):
-    if response_json is None or str(response_json.get("success")) != "True":
-        image_list, next_cursor, total_number_of_images = None, None, None
-    else:
-        image_list = response_json.get("items")
-        next_cursor = response_json.get("nextCursor")
-        total_number_of_images = response_json.get("totalResults")
+    def _get_timestamp_query_param(self, date):
+        date_obj = datetime.strptime(date, "%Y-%m-%d")
+        utc_date = date_obj.replace(tzinfo=timezone.utc)
+        start_timestamp = utc_date.isoformat()
+        end_timestamp = (utc_date + timedelta(days=1)).isoformat()
 
-    return image_list, next_cursor, total_number_of_images
+        start_timestamp = start_timestamp.replace("+00:00", "Z")
+        end_timestamp = end_timestamp.replace("+00:00", "Z")
 
+        return f"timestamp_created:[{start_timestamp} TO {end_timestamp}]"
 
-def _process_image_list(image_list):
-    prev_total = 0
-    total_images = 0
-    for image_data in image_list:
-        total_images = _process_image_data(image_data)
-        if total_images is None:
-            total_images = prev_total
-        else:
-            prev_total = total_images
-
-    return total_images
-
-
-def _process_image_data(image_data, sub_providers=SUB_PROVIDERS, provider=PROVIDER):
-    logger.debug(f"Processing image data: {image_data}")
-    license_url = _get_license_url(image_data.get("rights"))
-    image_url = image_data.get("edmIsShownBy")[0]
-    foreign_landing_url = _get_foreign_landing_url(image_data)
-    foreign_id = image_data.get("id")
-    title = image_data.get("title")[0]
-    meta_data = _create_meta_data_dict(image_data)
-
-    data_providers = set(meta_data["dataProvider"])
-    eligible_sub_providers = {
-        s for s in sub_providers if sub_providers[s] in data_providers
-    }
-    if len(eligible_sub_providers) > 1:
-        raise Exception(
-            f"More than one sub-provider identified for the "
-            f"image with foreign ID {foreign_id}"
-        )
-    source = (
-        eligible_sub_providers.pop() if len(eligible_sub_providers) == 1 else provider
-    )
+    def get_next_query_params(self, prev_query_params) -> dict:
+        if not prev_query_params:
+            return self.base_request_body
 
-    license_info = get_license_info(license_url=license_url)
+        return prev_query_params | {
+            "cursor": self.cursor,
+        }
 
-    return image_store.add_item(
-        foreign_landing_url=foreign_landing_url,
-        image_url=image_url,
-        license_info=license_info,
-        foreign_identifier=foreign_id,
-        title=title,
-        meta_data=meta_data,
-        source=source,
-    )
+    def get_should_continue(self, response_json):
+        if response_json.get("success") != "True":
+            logger.warning('Request failed with ``success = "False"``')
+            return False
```

Comment:
I _think_ this logic might be better suited in `get_batch_data`, since `get_should_continue` is typically concerned with the token/cursor iteration rather than error handling. @stacimc what do you think?

Reply:
What would you suggest should change here, specifically? Would `get_batch_data` just return an empty list in this case?

I don't personally see a big difference here... but I like that all the "is this response usable and can we move forward" is encapsulated in a single method rather than spread across two methods. But there may be a call order issue here, if "success" is False then we can't process the batch at all (in addition to not continuing, to maintain parity behaviour). :thinking:

Reply:
Yes, I would suggest moving L158 and L159 here to `get_batch_data`, and add a `return None` within the condition rather than `return False` (although `False` would work too because the check that follows is [falsy anyways](https://github.com/WordPress/openverse-catalog/blob/b13390fda17e14fc8b92559490a31eae80437d66/openverse_catalog/dags/providers/provider_api_scripts/provider_data_ingester.py#L173)). Here are some examples from other providers:

* [Brooklyn Museum](https://github.com/WordPress/openverse-catalog/blob/9be8bcec541f95dcd38d76a1064ce72cd75f8255/openverse_catalog/dags/providers/provider_api_scripts/brooklyn_museum.py#L41-L48)
* [Finnish Museum](https://github.com/WordPress/openverse-catalog/blob/1882d7776a16c6d7ebc9d0a28c502c25777c6a40/openverse_catalog/dags/providers/provider_api_scripts/finnish_museums.py#L47-L56)

My interpretation of these methods is that if we try to get the batch and there's an error, there's functionally no batch to process (as you mention) and `get_batch_data` should return `None`. The `get_should_continue` mechanism is for more peculiar cases (like this one, or Wikimedia) where the API _additionally_ provides some token or other mechanism which tells us whether we should continue processing.

Reply:
> My interpretation of these methods is that if we try to get the batch and there's an error, there's functionally no batch to process (as you mention) and get_batch_data should return None

Interesting! This makes sense, but then my instinct is that `get_batch_data` should raise an error in these cases that is caught and handled differently. Returning `None` seems ambiguous to me (though this is just my first time working on a provider script, so take that with a grain of salt).

I'll make the changes you suggested, thanks, Madison.

Reply:
For context on this, these scripts have historically operated in a manner that prioritizes gathering available records and preventing errors as much as possible. This was because (previously) an error condition meant that however x many hours of processing that had already occurred had to be done _again_ if we wanted to re-run a given day. We'd still get all the data we had gotten so far, but we hadn't had a mechanism until #650 (and with further improvement coming via #702) to pick up at a specific set of query params.

I think maintaining that resiliency within the scripts can be a good thing! However, particularly as we finally get everything back up and running we can start operating more under the principal of letting things break, because that tells us our assumptions about the API responses were wrong.